src/SymOp.jl

```diff
@@ -11,44 +11,45 @@
 # τ = -W^-1 w
 # (all these formulas are valid both in reduced and cartesian coordinates)
 
+# Time-reversal symmetries are the anti-unitaries
+# (Uu)(x) = conj(u(Wx+w))
+# or in Fourier space
+# (Uu)(G) = e^{i G τ} conj(u(-S^-1 G))
+
 # Tolerance to consider two atomic positions as equal (in relative coordinates)
 const SYMMETRY_TOLERANCE = 1e-5
 
-# Represents a symmetry (S,τ)
+# Represents a symmetry (W,w)
 struct SymOp{T <: Real}
-    S::Mat3{Int}
-    τ::Vec3{T}  # floating-point type fixed by spglib
-    function SymOp(S, τ)
-        τ = mod.(τ, 1)
-        new{eltype(τ)}(S, τ)
+    W::Mat3{Int}
+    w::Vec3{T}
+    function SymOp(W, w)
+        w = mod.(w, 1)
+        new{eltype(w)}(W, w)
     end
-    # compatibility with old stuff, will be removed at some point but doesn't hurt for now
-    SymOp(Sτ::Tuple) = SymOp(Sτ...)
 end
-
-function get_Ww(op::SymOp)
-    W = op.S'  # S = W'
-    w = -W*op.τ  # τ = -W^-1 w
-    W, w
+Base.propertynames(symop::SymOp, private::Bool=false) = ("W", "w", "S", "τ")
+function Base.getproperty(symop::SymOp, name::Symbol)
+    name == :S && return symop.W'
+    name == :τ && return -symop.W \symop.w
+    getfield(symop, name)
 end
 
-Base.:(==)(op1::SymOp, op2::SymOp) = op1.S == op2.S && op1.τ == op2.τ
+Base.:(==)(op1::SymOp, op2::SymOp) = op1.W == op2.W && op1.w == op2.w
 function Base.isapprox(op1::SymOp, op2::SymOp; atol=SYMMETRY_TOLERANCE)
     is_approx_integer(r) = all(ri -> abs(ri - round(ri)) ≤ atol, r)
-    op1.S == op2.S && is_approx_integer(op1.τ - op2.τ)
+    op1.W == op2.W && is_approx_integer(op1.w - op2.w)
 end
 Base.one(::Type{SymOp}) = SymOp(Mat3{Int}(I), Vec3(zeros(Bool, 3)))
 Base.one(::SymOp) = one(SymOp)
 
 # group composition and inverse.
-# Derived either from the formulas for the composition/inverse of W/w
-# then passing to reciprocal, or directly from the symmetry operation in reciprocal space
 function Base.:*(op1, op2)
-    S = op1.S * op2.S
-    τ = op1.τ + op1.S' \ op2.τ
-    SymOp(S, τ)
+    W = op1.W * op2.W
+    w = op1.w + op1.W * op2.w
+    SymOp(W, w)
```

Comment:
Make this a one-liner, too?

Reply:
these are non-trivial formulas, better as separate lines